### ISSUE TITLE
./bin/magento config:show fails with a fatal error

### ISSUE THREAD

**Issue by simonworkhouse** [magento/magento2#17582]
The ./bin/magento config:show command fails with a fatal error after running ./bin/magento app:config:dump
Similar issue reported previously https://github.com/magento/magento2/issues/16654

### Preconditions

1. Magento 2.2.5
2. PHP 7.1.20
3. MySQL 5.7.23

### Steps to reproduce

1. composer create-project --repository-url=https://repo.magento.com/ magento/project-community-edition {install dir}
2. cd {install dir}
3. ./bin/magento setup:install --backend-frontname=? --db-host=? --db-name=? --db-user=? --db-password=? --base-url=? --admin-user=? --admin-password=? --admin-email=? --admin-firstname=? --admin-lastname=?
4. ./bin/magento app:config:dump
5. ./bin/magento config:show

### Expected result

All config data output correctly without a fatal error.

### Actual result

...
catalog/productalert_cron/error_email - 
catalog/product_video/play_if_base - 0
catalog/product_video/show_related - 0
catalog/product_video/video_auto_restart - 0
catalog/review/allow_guest - 1
catalog/search/engine - mysql
catalog/search/min_query_length - 1
catalog/search/max_query_length - 128
catalog/search/max_count_cacheable_search_terms - 100
analytics/subscription/enabled - 1
PHP Fatal error:  Uncaught Error: Call to undefined method Magento\Config\Model\Config\Structure\Element\Group\Interceptor::hasBackendModel() in <install dir>/vendor/magento/module-config/Console/Command/ConfigShow/ValueProcessor.php:100
Stack trace:
\#0 <install dir>/vendor/magento/module-config/Console/Command/ConfigShowCommand.php(197): Magento\Config\Console\Command\ConfigShow\ValueProcessor->process('default', '', 'Magento Analyti...', 'analytics/integ...')
\#1 <install dir>/vendor/magento/module-config/Console/Command/ConfigShowCommand.php(202): Magento\Config\Console\Command\ConfigShowCommand->outputResult(Object(Symfony\Component\Console\Output\ConsoleOutput), 'Magento Analyti...', 'analytics/integ...')
\#2 <install dir>/vendor/magento/module-config/Console/Command/ConfigShowCommand.php(202): Magento\Config\Console\Command\ConfigShow in <install dir>/vendor/magento/module-config/Console/Command/ConfigShow/ValueProcessor.php on line 100

**Comment by magento-engcom-team** [comment]
Hi @simonworkhouse. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me {$VERSION} instance` 

where `{$VERSION}` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (2.2-develop +). 
For more details, please, review the [Magento Contributor Assistant](https://github.com/magento/magento2/wiki/Magento-Contributor-Assistant) documentation.

@simonworkhouse do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by ghost** [comment]
@simonworkhouse , thank you for your report.
We've acknowledged the issue and added to our backlog.

**Comment by leoquijano** [comment]
Hi,

I also encountered this issue. My Magento version is v2.2.5. In my case, this happens when ```config:show``` tries to display the value of ```dashboard/use_aggregated_data``` from ```app/etc/config.php```:

```php
'system' => [
  'default' => [
     // ...
     'dashboard' => [
        'use_aggregated_data' => '0',
     ],
     // ...
  ]
]
```

It seems that ```Magento\Config\Console\Command\ConfigShow::process``` is assuming that the ```Magento\Config\Model\Config\Structure::getElementByConfigPath``` method will always return an instance of ```Magento\Config\Model\Config\Structure\Element\Field```:

```php
public function process($scope, $scopeCode, $value, $path)
{
    // ...
    /** @var Field $field */
    $field = $configStructure->getElementByConfigPath($path);

    /** @var Value $backendModel */
    $backendModel = $field && $field->hasBackendModel()
        ? $field->getBackendModel()
        : $this->configValueFactory->create();

    // ...
}
```

However, it actually returns an instance of ```\Magento\Config\Model\Config\Structure\ElementInterface``` (note the ```@return``` PHPDoc tag):

```php
/**
 * Find element by config path
 *
 * @param string $path The configuration path
 * @return \Magento\Config\Model\Config\Structure\ElementInterface|null
 * @since 100.2.0
 */
public function getElementByConfigPath($path)
{
    $allPaths = $this->getFieldPaths();

    if (isset($allPaths[$path])) {
        $path = array_shift($allPaths[$path]);
    }

    return $this->getElementByPathParts(explode('/', $path));
}
```

In the case of ```dashboard/use_aggregated_data```, the returned element is actually an instance of ```Magento\Config\Model\Config\Structure\Element\Group```, which doesn't have the ```hasBackendModel``` method. This triggers a fatal error by PHP.

A fix could be either adjusting the ```process``` field to handle group elements, or making ```getElementByConfigPath``` only return field elements. A quick workaround while the bug gets fixed is to remove the offending entry from ```app/etc/config```:

```php
 'dashboard' => [
    'use_aggregated_data' => '0',
 ],
```

If the entry is required to be there, then a temporary adjustment to the ```process``` method above might help.

**Comment by max-a-mdmitriev** [comment]
#khcd2018

**Comment by magento-engcom-team** [comment]
@max-a-mdmitriev thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by keyurshah070** [comment]
I am working on #mm18pl

**Comment by magento-engcom-team** [comment]
@keyurshah070 thank you for joining. Please accept team invitation [here](https://github.com/orgs/magento/invitation) and self-assign the issue.

**Comment by magento-engcom-team** [comment]
Hi @simonworkhouse. Thank you for your report. 
The issue has been fixed in magento/magento2#17993 by @keyurshah070 in 2.2-develop branch
Related commit(s):
 - [ae503f60a0e07b1611076802a6b219b2c1dc5bb4](https://github.com/magento/magento2/commit/ae503f60a0e07b1611076802a6b219b2c1dc5bb4)

The fix will be available with the upcoming 2.2.8 release.

**Comment by slavvka** [comment]
Hi @simonworkhouse. Thank you for your report. 
The issue has been fixed in magento/magento2#18295 by @mage2pratik in 2.3-develop branch
Related commit(s):
 - [20af5a9cbd21b55cb432110ef2a44db0b5156b44](https://github.com/magento/magento2/commit/20af5a9cbd21b55cb432110ef2a44db0b5156b44)

The fix will be available with the upcoming 2.3.1 release.
